promiseFns/filter.js

```diff
@@ -1,6 +1,9 @@
 const util = require("./util");
 module.exports = (Bluebird) => {
-    Bluebird.filter = (x, predicate, opts) => Bluebird.resolve(x).filter(predicate, opts);
```

Comment:
Because you resolve before you do the filter, you need to resolve the filter values.